client/server/middleware/assets.js

```diff
@@ -26,20 +26,27 @@ const getChunkById = ( assets, chunkId ) => assets.chunks.find( ( chunk ) => chu
 const groupAssetsByType = ( assets ) => defaults( groupBy( assets, getAssetType ), EMPTY_ASSETS );
 
 export default () => {
-	return ( req, res, next ) => {
-		req.getAssets = () => {
-			return JSON.parse( fs.readFileSync( getAssetsPath(), 'utf8' ) );
-		};
+	let assetsFile;
+	async function readAssets() {
+		if ( ! assetsFile ) {
+			assetsFile = JSON.parse( await readFile( ASSETS_FILE, 'utf8' ) );
+		}
+		return assetsFile;
+	}
+
+	return asyncHandler( async ( req, res, next ) => {
```

Comment:
In my view the services provided by `express-async-handler` are too trivial to warrant a new dependency -- it basically adds a `.catch( next )` handler and that's all. The same can be written as:
```js
return async ( req, res, next ) => {
  try {
    const assets = await readAssets();
    req.getAssets = ...
    req.getFilesForEntrypoint = ...
    next();
  } catch ( err ) {
    next( err );
  }
}
```
But your version works, too.